Controller/SourceController.php

```diff
@@ -45,6 +45,12 @@ public function getAction($source) {
         $sourceConfig = $config['sources'][$source];
         /** @var AnalyticsInterface $analytics */
         $analytics = new $sourceConfig['class']($this->get('service_container'));
+        if (method_exists($analytics, 'setContext')) {
```

Comment:
Also, is there ever a case where $analytics won't have this method? If so, this is confusing. We should limit our use of duck-typing.